PyStemmusScope/stemmus_scope.py

```diff
@@ -94,23 +155,34 @@ def run(self) -> str:
         Args:
 
         Returns:
-            Tuple with stdout and stderr
+            string, the model log
         """
-
-        # run the model
-        args = [f"{self.exe_file} {self.cfg_file}"]
-        result = subprocess.run(
-            args, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True, check=True,
-        )
-        stdout = result.stdout
-
-        # TODO return log info line by line!
-        logger.info("%s", stdout)
-
-        return stdout
+        if self.exe_file:
+            # run using MCR
+            args = [f"{self.exe_file} {self.cfg_file}"]
+            # set matlab log dir
+            os.environ['MATLAB_LOG_DIR'] = str(self._config["InputPath"])
+            result = _run_sub_process(args, None)
+        if self.sub_process=="Matlab":
+            # set Matlab arguments
+            path_to_config = f"'{self.cfg_file}'"
+            command_line = f'matlab -r "STEMMUS_SCOPE_exe({path_to_config});exit;"'
+            args = [command_line, "-nodisplay", "-nosplash", "-nodesktop"]
+            result = _run_sub_process(args, self.model_src)
+        if self.sub_process=="Octave":
+            # set Octave arguments
+            # use subprocess instead of oct2py,
+            # see issue STEMMUS_SCOPE_Processing/issues/46
+            path_to_config = f"'{self.cfg_file}'"
+            # fix for windows
+            path_to_config = path_to_config.replace("\\", "/")
+            command_line = f'octave --eval "STEMMUS_SCOPE_exe({path_to_config});exit;"'
+            args = [command_line, "--no-gui", "--silent"]
```

Comment:
```suggestion
            eval_code = f'STEMMUS_SCOPE_exe({path_to_config});exit;'
            args = ["octave", "--eval", eval_code, "--no-gui", "--silent"]
```

This still has to be changed for it to work on Windows. All the arguments have to be passed separately to subprocess for it to work.

Reply:
ah sorry. I commit your suggestion here, but I will check if the code works on linux too.